### PR TITLE
Manage @template tags

### PR DESCRIPTION
**Pull request by osechet**
This PR adds management of @template tags by adding the defined templates to the class name and method name.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,9 +11,11 @@
   },
   "devDependencies": {
     "@englercj/code-style": "^1.0.6",
-    "eslint": "^3.3.1"
-  },
-  "peerDependencies": {
+    "eslint": "^3.3.1",
     "jsdoc": "^3.4.0"
```
Comment (englercj) [inline package.json]: jsdoc should remain a peer dependency, otherwise it will not notify users they need it when they install this module.

**Comment by osechet** [comment]
Oups, I never noticed `@template` was not standard. It's in fact provided by closure compiler and used by several js libraries. I don't think it's a good thing to create a new `@generic` tag. It would be better to have a mechanism to say we want the support of closure compiler's tags.

**Comment by englercj** [comment]
Ah! Didn't realize it was a closure compiler tag, that makes more sense. In that case, can you just add a section mentioning supported CC tags and adding a link for that one to the docs for that tag?

**Comment by englercj** [comment]
Merged this, will manually fix the few issues I had.

**Pull request merged**